### PR TITLE
Add FPM variants (styled after OwnCloud's script updates for this same thing)

### LINKED ISSUE #17: Add tag for PHP FPM
Apache with PHP FPM is much better than mod_php. How about to provide alternative?

### PR DESCRIPTION
**Pull request by tianon**
Closes #17
Closes #31 (adapted from)

cc @iBobik @bluefoxicy

### CODE REVIEW

**Comment by tianon** [comment]
```diff
diff --git a/7/apache/Dockerfile b/7/fpm/Dockerfile
index f61607e..b6675d8 100644
--- a/7/apache/Dockerfile
+++ b/7/fpm/Dockerfile
@@ -1,8 +1,6 @@
 # from https://www.drupal.org/requirements/php#drupalversions
 FROM php:5.6-apache
 
-RUN a2enmod rewrite
-
 # install the PHP extensions we need
 RUN apt-get update && apt-get install -y libpng12-dev libjpeg-dev libpq-dev \
 	&& rm -rf /var/lib/apt/lists/* \
```
```diff
diff --git a/8/apache/Dockerfile b/8/fpm/Dockerfile
index 212ec74..fe010d2 100644
--- a/8/apache/Dockerfile
+++ b/8/fpm/Dockerfile
@@ -1,7 +1,5 @@
 # from https://www.drupal.org/requirements/php#drupalversions
-FROM php:5.6-apache
-
-RUN a2enmod rewrite
+FROM php:5.6-fpm
 
 # install the PHP extensions we need
 RUN apt-get update && apt-get install -y libpng12-dev libjpeg-dev libpq-dev \
```

**Comment by tianon** [comment]
```diff
diff --git a/owncloud/generate-stackbrew-library.sh b/drupal/generate-stackbrew-library.sh
index 27cc015..5c37295 100755
--- a/owncloud/generate-stackbrew-library.sh
+++ b/drupal/generate-stackbrew-library.sh
@@ -3,26 +3,21 @@ set -e
 
 declare -A aliases
 aliases=(
-	[8.2]='8 latest'
-	[7.0]='7'
-	[6.0]='6'
+	[8]='latest'
 )
 
 cd "$(dirname "$(readlink -f "$BASH_SOURCE")")"
 
 versions=( */ )
 versions=( "${versions[@]%/}" )
-url='git://github.com/docker-library/owncloud'
+url='git://github.com/docker-library/drupal'
 
 echo '# maintainer: InfoSiftr <github@infosiftr.com> (@infosiftr)'
 
-echo
-echo '# https://github.com/owncloud/core/wiki/Maintenance-and-Release-Schedule'
-
 for version in "${versions[@]}"; do
 	for variant in apache fpm; do
 		commit="$(cd "$version/$variant" && git log -1 --format='format:%H' -- Dockerfile $(awk 'toupper($1) == "COPY" { for (i = 2; i < NF; i++) { print $i } }' Dockerfile))"
-		fullVersion="$(grep -m1 'ENV OWNCLOUD_VERSION ' "$version/$variant/Dockerfile" | cut -d' ' -f3)"
+		fullVersion="$(grep -m1 'ENV DRUPAL_VERSION ' "$version/$variant/Dockerfile" | cut -d' ' -f3)"
 
 		versionAliases=()
 		while [ "$fullVersion" != "$version" -a "${fullVersion%[.-]*}" != "$fullVersion" ]; do
```

File: 7/fpm/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# from https://www.drupal.org/requirements/php#drupalversions
+FROM php:5.6-apache
```
Comment (skyred) [inline 7/fpm/Dockerfile]: Is this a typo? Should be `FROM php:5.6-fpm`

**Comment by skyred** [comment]
Also, shall we take this chance to switch to PHP 7 for Drupal? 

Here is why: http://fabianx.drupalgardens.com/blog/why-drupal-8-defaulting-its-testing-php7-and-why-you-should-too

And the current status of PHP 7 Support in D8: 

> Drupal 8 currently has 100% pass on PHP 7
https://www.drupal.org/node/2454439

**Comment by iBobik** [comment]
About Drupal 8, PHP 7 is ok, but not for D7 yet.



Jan Pobořil

2016-02-04 4:41 GMT+01:00 Jingsheng Wang <notifications@github.com>:

**Comment by yosifkit** [comment]
@tianon, this needs a rebase :wink:.

File: 7/fpm/Dockerfile (hunk 1 above)
Comment (tianon) [inline 7/fpm/Dockerfile]: Nice catch, fixed! :+1:

**Comment by tianon** [comment]
Rebased -- I think it is appropriate to have the PHP 7 discussion orthogonal to this since consistency is important so we're not going to switch PHP versions just for FPM.

**Comment by yosifkit** [comment]
LGTM

**Pull request merged**